packages/core/src/Nodes/nodeFactory.ts

```diff
@@ -0,0 +1,106 @@
+import { Registry } from '../Registry';
+import { Socket } from '../Sockets/Socket';
+import { EventNode2 } from './EventNode';
+import { FlowNode2 } from './FlowNode';
+import { NodeConfiguration } from './Node';
+import { Node } from './Node';
+import {
+  EventNodeDefinition,
+  IFlowNodeDefinition,
+  INodeDefinitionBase,
+  NodeCategory
+} from './NodeDefinition';
+import { NodeDescription2 } from './Registry/NodeDescription';
+
+type AnyInOut = Record<string, string>;
+
+function isFlowNode(
+  nodeDefinition: INodeDefinitionBase
+): nodeDefinition is IFlowNodeDefinition<AnyInOut, AnyInOut, AnyInOut> {
+  if (nodeDefinition.category === NodeCategory.Flow) {
```

Comment:
For this cases you can avoid the condition and just return the check
```suggestion
  return nodeDefinition.category === NodeCategory.Flow;
```

Reply:
> @oveddan sorry to bother you one more time, I'm giving a try to the `FunctionDesc` implementation and I'm having questions during the process. One thing I noticed about the inputs definition, is there's no way to define default values and choices, does this PR fix that?
> 
> Checking the previous implementations for the Log node, we should be able to define this kind of inputs
> 
> ```json
> "in": {
>   "text": "string",
>   "severity": {
>     "valueType": "string",
>     "defaultValue": "info",
>     "choices": ["verbose", "info", "warning", "error"]
>   }
> }
> ```

no problem at all about your bothering, the feedback is great.

being able to define choices would be great and I see now it was missing.
defaultValue is currently defined in a separate property: `initialInputsVals`

However I do think you are right that for each socket, we should have an object that can define these things;  I'll modify accordingly

Reply:
This has been fixed now; for each socket, you can define either a string, or an object with properties:

```typescript
export interface SocketDefinition {
  valueType: string;
  defaultValue?: any;
  choices?: string[];
}

export type SocketsDefinition = Record<string, SocketDefinition | string>;

export interface INodeDefinitionBase {
  category: NodeCategory;
  typeName: string;
  otherTypeNames?: string[];
  // type: NodeType;
  aliases?: string[]; // for backwards compatibility
  helpDescription?: string;
  label?: string;
  in: SocketsDefinition;
  out: SocketsDefinition;
}
```